File: src/reproducible.R
library( ANTsR )
fi <- antsImageRead(getANTsRData("r16") )
mi <- antsImageRead(getANTsRData("r64") )
tx = 'SyN'
metrics = c( 'MeanSquares','GC',"Mattes")
for ( metric in metrics ) {
  mytx1 <- antsRegistration(fixed=fi, mi, typeofTransform = tx, affMetric=metric )
  p1=getAntsrTransformParameters( readAntsrTransform( mytx1$fwdtransforms[2] ))
  for ( k in 1:1 ) {
    mytx2 <- antsRegistration(fixed=fi, mi, typeofTransform = tx, affMetric=metric  )
    p2=getAntsrTransformParameters( readAntsrTransform( mytx2$fwdtransforms[2] ))
    print( paste( metric, "Change", sum( abs( p1 - p2 ) ), "Similarity", antsImageMutualInformation( fi,mytx2$warpedmovout) ) )
  }
}

File: src/reproducible.py
import ants
fi = ants.image_read( ants.get_data( "r16") )
mi = ants.image_read( ants.get_data( "r64") )
tx = 'SyN'
metrics = [ "MeanSquares", "GC", "Mattes" ]
asr = 0.1
for m in range(0, len(metrics)):
  mytx1 = ants.registration(fixed=fi, moving=mi, type_of_transform = tx, aff_metric=metrics[m], aff_random_sampling_rate=asr )
  p1 = ants.read_transform( mytx1['fwdtransforms'][1] )
  p1 = ants.get_ants_transform_parameters( p1 )
  for k in range(0,2):
    mytx2 = ants.registration(fixed=fi, moving=mi, type_of_transform = tx, aff_metric=metrics[m], aff_random_sampling_rate=asr  )
    p2 = ants.read_transform( mytx2['fwdtransforms'][1] )
    p2 = ants.get_ants_transform_parameters( p2 )
    metval = ants.image_mutual_information( fi, mytx2['warpedmovout'] )
    chg = abs(p2-p1).sum()
    print( metrics[m] + " Change: try " + str(k) + " Is " + str( chg ) + " Similarity= " + str(metval) )
